### PR TITLE
rejigger the demo with examples that actually make sense

### LINKED ISSUE #56: demonstrate binding form invalid state to submit button disabled state
A common UI pattern is to auto-validate form inputs as the user is filling out the form, and to disable the submit button until the whole form is valid, for a more responsive experience.

Both of the examples in the [demo](https://elements.polymer-project.org/elements/iron-form?view=demo:demo/index.html&active=iron-form) currently show off validating the form when the submit button is clicked. They demonstrate slightly different inputs as well as GET vs POST, but the second example doesn't demonstrate anything you couldn't figure out from the first.

It would be much more illustrative if one of the examples showed off `<paper-input>`s with `auto-validate`, and had its submit button's disabled state change to reflect the form's validity responsively. As a newcomer to Polymer, the intended (and cleanest) way to wire this behavior together is not clear.

Would you consider adapting the demo to show off this use case?

Thanks for your consideration and for the great work on Polymer.

### LINKED ISSUE #68: demonstrate reacting to form's submitted state changes
Ajax forms transition between submitted states like this:
not submitted → submitted / request in flight → (response received, so back to) not submitted

A common UI pattern is to disable the submit button and change its text from "submit" to "loading..." (show a spinner, etc.) while the request is in flight.

An iron-form is submitted via an internal iron-ajax element. iron-ajax has a "loading" property for whether a request is in flight. Does iron-form expose this property so the pattern above can be expressed simply? Or do we have to duplicate the logic that keeps track of the "loading" state on the form itself?

Would you consider demonstrating this pattern in the iron-form demo page to show off the recommended way to do this? Thanks for your consideration.

### PR DESCRIPTION
**Pull request by notwaldorf**
Fixes #56, fixes #68

/cc @robdodson 

screen shot 2016-02-23 at 4 44 08 pm
screen shot 2016-02-23 at 4 44 15 pm
screen shot 2016-02-23 at 4 44 23 pm
screen shot 2016-02-23 at 4 44 29 pm
screen shot 2016-02-23 at 4 44 35 pm

### CODE REVIEW

**Comment by notwaldorf** [comment]
@valdrinkoshi PTAL

**Comment by robdodson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by robdodson** [comment]
left some comments in slack, LGTM after those get fixed up :D

**Comment by notwaldorf** [comment]
@robdodson donzo!

**Comment by robdodson** [comment]
LGTM!

File: demo/index.html [hunk 1]
```diff
@@ -17,35 +17,67 @@
...
+  <link rel="import" href="../../paper-menu/paper-menu.html">
+  <link rel="import" href="../../paper-spinner/paper-spinner.html">
+  <link rel="import" href="../../paper-styles/color.html">
+  <link rel="import" href="../../paper-styles/typography.html">
   <link rel="import" href="../iron-form.html">
   <link rel="import" href="simple-element.html">
+
+  <style is="custom-style" include="demo-pages-shared-styles">
+    form {
+      width: 100%;
```
Comment (valdrinkoshi) [inline demo/index.html]: It seems this one is not needed (renders correctly without it)

**Comment by valdrinkoshi** [comment]
I think there is an extra style that is not needed but also doesn't hurt. 
LGTM

**Pull request merged**